bin/pycbc_optimize_snr

```diff
@@ -286,67 +60,36 @@ parser.add_argument('--output-path', required=True,
-                    help='Only relevant for --optimizer differential_evolution: '
-                    'The maximum number of generations over which the entire '
-                    'population is evolved.')
-parser.add_argument('--di-popsize', type=int, default=100,
-                    help='Only relevant for --optimizer differential_evolution: '
-                         'A multiplier for setting the total population size.')
-parser.add_argument('--shgo-samples', type=int, default=76,
-                    help='Only relevant for --optimizer shgo: '
-                    'Number of sampling points used in the construction of the '
-                    'simplicial complex.')
-parser.add_argument('--shgo-iters', type=int, default=3,
-                    help='Only relevant for --optimizer shgo: '
-                    'Number of iterations used in the construction of the simplicial complex.')
-parser.add_argument('--pso-iters', type=int, default=5,
-                    help='Only relevant for --optimizer pso: '
-                    'Number of iterations used in the particle swarm optimization.')
-parser.add_argument('--pso-particles', type=int, default=250,
-                    help='Only relevant for --optimizer pso: '
-                    'Number of particles used in the swarm.')
-parser.add_argument('--pso-c1', type=float, default=0.5,
-                    help='Only relevant for --optimizer pso: '
-                    'The hyperparameter c1: the cognitive parameter.')
-parser.add_argument('--pso-c2', type=float, default=2.0,
-                    help='Only relevant for --optimizer pso: '
-                    'The hyperparameter c2: the social parameter.')
-parser.add_argument('--pso-w', type=float, default=0.01,
-                    help='Only relevant for --optimizer pso: '
-                    'The hyperparameter w: the inertia parameter.')
-parser.add_argument('--include-candidate-in-optimizer', action='store_true',
-                    help='Include parameters of the candidate event in the '
-                    'initialised array for the optimizer. Only relevant for '
-                    '--optimizer pso or differential_evolution')
-parser.add_argument('--seed', type=int, default=42,
-                    help='Seed to supply to the random generation of initial '
-                    'array to pass to the optimizer. Only relevant for '
-                    '--optimizer pso or differential_evolution')
-
+snr_optimizer.insert_snr_optimizer_options(parser)
 scheme.insert_processing_option_group(parser)
 fft.insert_fft_option_group(parser)
 
-# Input checking
 args = parser.parse_args()
-if args.snr_threshold != 4:
-    parser.error("Sorry, the SNR threshold option doesn't work yet")
-if args.optimizer == 'pso' and ps == None:
-    parser.error('You need to install pyswarms to use the pso optimizer.')
+
 pycbc.init_logging(args.verbose)
 
-if args.seed:
-    numpy.random.seed(args.seed)
+if args.optimizer_seed:
```

Comment:
What if I want to use a seed of zero? This should be `if args.optimizer_seed is not None:`

Reply:
ah very true, I forget that python `0 == False`!

Reply:
In fact the current default is 42, so we may not even need the if statement

Reply:
See my suggestion. (Basically allow 'None' or a string which can be converted to integer, and the default answer should be '42')

Reply:
Okay, so you would specifically state `--optimizer-seed None` in the arguments if you want it turned off?

Reply:
Yes, that's what I said. 'None' would access the default behaviour of numpy.random.seed when given None ... maybe 'random' would be a good alias for that